### PR TITLE
Upgrade solc

### PR DESCRIPTION
**Pull request by u2**
https://github.com/cryptape/cita-build/issues/8

Upgrade solc to 0.4.22

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -32,17 +34,23 @@ RUN pip3 install -U pip \
      && pip3 install git+https://github.com/ethereum/pyethereum.git@3d5ec14032cc471f4dcfc7cc5c947294daf85fe0 \
      && rm -r ~/.cache/pip
 
-COPY solc /usr/bin/
-RUN chmod +x /usr/bin/solc
+RUN curl -o solc -L https://github.com/ethereum/solidity/releases/download/v0.4.24/solc-static-linux \
+  && mv solc /usr/bin/ \
+  && chmod +x /usr/bin/solc
```
Comment (TheWaWaR) [inline Dockerfile]: Are we upgrade `solc` version here?

Our currently used version is `0.4.19`

**Comment by TheWaWaR** [comment]
Can all tests dependent on `solc` still works?

**Comment by kaikai1024** [comment]
LIke python, we can keep 0.4.19 and 0.4.24, then link the solc as 0.4.19 or 0.4.24. User can choose to use solc-0.4.23 or solc(0.4.19)

**Comment by u2** [comment]
But if we upgrade the solc version in cita, there will be only one choice. @kaikai1024

**Comment by u2** [comment]
@TheWaWaR  yes, the PR #992

**Review by TheWaWaR — APPROVED** [review]

**Pull request merged**